## Great America PAC May Monthly FEC Filing

Great America PAC, a super PAC backing Donald Trump for President, filed its [latest report](http://docquery.fec.gov/cgi-bin/forms/C00608489/1073787/) with the Federal Election Commission on May 20. Here are some details from it, thanks to [agate](https://agate.readthedocs.io/en/1.3.1/).

In [1]:
import agate

In [2]:
contributions = agate.Table.from_csv('great_america_sa.csv')

In [3]:
print contributions

|--------------------+---------------|
|  column_names      | column_types  |
|--------------------+---------------|
|  linenumber        | Text          |
|  fec_committee_id  | Text          |
|  transaction_id    | Text          |
|  entity_type       | Text          |
|  organization_name | Boolean       |
|  last_name         | Text          |
|  first_name        | Text          |
|  middle_name       | Boolean       |
|  prefix            | Text          |
|  suffix            | Boolean       |
|  street_1          | Text          |
|  street_2          | Text          |
|  city              | Text          |
|  state             | Text          |
|  zip_code          | Number        |
|  election_code     | Boolean       |
|  date              | Date          |
|  amount            | Number        |
|  aggregate_amount  | Number        |
|  employer          | Text          |
|  occupation        | Text          |
|  memo_code         | Boolean       |
|  memo_text         | Te

Let's find the largest contribution and who gave it:

In [5]:
print contributions.aggregate(agate.Max('amount'))

50000.0


In [7]:
top_donor = contributions.where(lambda row: row['amount'] == 50000.0)

In [13]:
print top_donor.rows[0]['last_name'] +', '+ top_donor.rows[0]['first_name'] + ' - ' + top_donor.rows[0]['city'] + ', ' + top_donor.rows[0]['state'] 

BEACH, ERIC - COSTA MESA, CA


How about the total number of itemized contributions - those that total at least $200 in aggregate from a donor:

In [15]:
print contributions.aggregate(agate.Count('transaction_id'))

125


Average itemized contribution: 

In [16]:
print contributions.aggregate(agate.Median('amount'))

300.0


One thing I noticed from browsing the [contributions](https://projects.propublica.org/itemizer/filing/1073787/schedule/sa) was the number of records that had no employer and occupation information, as required by law. These are indicated by the campaign listing "Information Requested Per Best Efforts" in the data. Let's look at those - how many are there?

In [17]:
no_emp_occ = contributions.where(lambda row: row['employer'] == 'INFORMATION REQUESTED PER BEST EFFORTS')

In [18]:
print no_emp_occ.aggregate(agate.Count('transaction_id'))

90


So 90 of the 125 itemized contributions in this filing contain no employer or occupation information about the donors. In percentage terms, that's: 

In [1]:
print (90.0/125.0)*100

72.0
